nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/pom.xml

```diff
@@ -204,181 +215,42 @@
             <activation>
                 <activeByDefault>false</activeByDefault>
             </activation>
+            <properties>
+                <elastic_docker_image>7.17.6</elastic_docker_image>
+            </properties>
             <build>
                 <pluginManagement>
                     <plugins>
                         <plugin>
                             <groupId>org.apache.maven.plugins</groupId>
                             <artifactId>maven-failsafe-plugin</artifactId>
+                            <configuration>
+                                <systemPropertyVariables>
+                                    <elastic.docker.image>docker.elastic.co/elasticsearch/elasticsearch:${elastic_docker_image}</elastic.docker.image>
+                                </systemPropertyVariables>
+                            </configuration>
                         </plugin>
                     </plugins>
                 </pluginManagement>
             </build>
         </profile>
-
-        <profile>
-            <!-- use with elasticsearch-oss only (-default can be used if x-pack-ml permissions fixed) -->
-            <id>elasticsearch-6</id>
-            <activation>
-                <activeByDefault>false</activeByDefault>
-            </activation>
-            <properties>
-                <es.int.version>6.8.23</es.int.version>
-                <es.int.type.name>_doc</es.int.type.name>
-                <es.int.script.name>setup-6.script</es.int.script.name>
-                <es.int.path.conf />
-            </properties>
-        </profile>
         <profile>
-            <!-- use with elasticsearch-oss or elasticsearch-default -->
-            <id>elasticsearch-7</id>
-            <activation>
-                <activeByDefault>false</activeByDefault>
-            </activation>
+            <id>elastic5</id>
             <properties>
-                <es.int.version>7.10.2</es.int.version>
-                <es.int.script.name>setup-7.script</es.int.script.name>
-                <es.int.type.name />
-                <es.int.path.conf />
+                <elastic_docker_image>5.6</elastic_docker_image>
```

Comment:
Being specific and using the latest versions of each major release would seem sensible (unless there are good reason to use less specific/older versions):

- 5.6.16
- 6.8.23
- 7.17.6
- 8.4.3

Are there any licensing and/or X-Pack concerns around using the newer images?

Do we need to cater for `arm64` machines running these tests (the later 7.x and 8.x images have specific `-arm64` image variants, I'd guess either 6.x and 5.x would either work or have to be skipped based on the user's machine architecture)

Reply:
I'm not sure about the arm64, but that shouldn't be a blocker because arm64 support in NiFi is still a work in progress. As for the licensing issues, I don't see why we would since elastic makes them freely available for demonstration and evaluation purposes. We're also not redistributing them or anything like that.